hazelcast/src/main/java/com/hazelcast/util/executor/StripedExecutor.java

```diff
@@ -45,20 +47,27 @@
     public static final AtomicLong THREAD_ID_GENERATOR = new AtomicLong();
 
     private final int size;
+    private final ILogger logger;
     private final Worker[] workers;
     private final Random rand = new Random();
-    private final int maximumQueueSize;
-    private final ILogger logger;
+
     private volatile boolean live = true;
 
-    public StripedExecutor(ILogger logger, String threadNamePrefix, ThreadGroup threadGroup
-            , int threadCount, int maximumQueueSize) {
+    public StripedExecutor(ILogger logger, String threadNamePrefix, ThreadGroup threadGroup,
+                           int threadCount, int maximumQueueCapacity) {
+        checkPositive(threadCount, "threadCount should be positive but found " + threadCount);
+        checkPositive(maximumQueueCapacity, "maximumQueueCapacity should be positive but found " + maximumQueueCapacity);
+
         this.logger = logger;
-        this.maximumQueueSize = maximumQueueSize;
         this.size = threadCount;
         this.workers = new Worker[threadCount];
+
+        // Max event queue capacity is determined via system property. Each worker in this executor should consume
```

Comment:
I don't think having *event queue* specific comment in `StripedExecutor ` is a good idea. `StripedExecutor` is not specific to event system only.